### ISSUE TITLE
Django 3.2 cache compatibility

### ISSUE THREAD

**Issue by wamberg** [jazzband/django-dbtemplates#116]
While using Django 3.2, `dbtemplates` throws the following error:

```
ImportError: cannot import name '_create_cache' from 'django.core.cache'
```

The error stems from `dbtemplates` cache wrapper:

https://github.com/jazzband/django-dbtemplates/blob/ac86ca5ec97c47016146b4c3d7b3f0c77ce4f614/dbtemplates/utils/cache.py#L12

Side note of gratitude, this library has come in very handy for us at PBS. Thank you! :bow:

**Comment by gamecat10** [comment]
Temporary fix:

```
--- ./dbtemplates/utils/cache.py.old	2021-04-06 17:18:12.741841778 +0300
+++ ./dbtemplates/utils/cache.py	2021-04-06 17:18:08.721937235 +0300
@@ -1,4 +1,5 @@
 from django.core import signals
+from django.core.cache import caches
 from django.contrib.sites.models import Site
 from django.template.defaultfilters import slugify
 
@@ -9,13 +10,7 @@
     """
     Compatibilty wrapper for getting Django's cache backend instance
     """
-    from django.core.cache import _create_cache
-    cache = _create_cache(settings.DBTEMPLATES_CACHE_BACKEND)
-    # Some caches -- python-memcached in particular -- need to do a cleanup at
-    # the end of a request cycle. If not implemented in a particular backend
-    # cache.close is a no-op
-    signals.request_finished.connect(cache.close)
-    return cache
+    return caches[settings.DBTEMPLATES_CACHE_BACKEND]
```

**Comment by ArnoldJudas** [comment]
my fix for support for django 3.2 without removing support for older versions
file: dbtemplates/utils/cache.py

```
--- 1	2021-04-14 10:54:09.984460911 +0200
+++ 2	2021-04-14 10:54:25.011612579 +0200
@@ -1,3 +1,4 @@
+import django
 from django.core import signals
 from django.contrib.sites.models import Site
 from django.template.defaultfilters import slugify
@@ -9,8 +10,12 @@
     """
     Compatibilty wrapper for getting Django's cache backend instance
     """
-    from django.core.cache import _create_cache
-    cache = _create_cache(settings.DBTEMPLATES_CACHE_BACKEND)
+    if django.VERSION[0] >= 3 and django.VERSION[1] >= 2:
+        from django.core.cache import caches
+        cache = caches.create_connection(settings.DBTEMPLATES_CACHE_BACKEND)
+    else:
+        from django.core.cache import _create_cache
+        cache = _create_cache(settings.DBTEMPLATES_CACHE_BACKEND)
     # Some caches -- python-memcached in particular -- need to do a cleanup at
     # the end of a request cycle. If not implemented in a particular backend
     # cache.close is a no-op

```

**Comment by jezdez** [comment]
Hey all, to move this forward it's best to just open PRs that completely pass the tests.

**Comment by fjbardelli** [comment]
Please PR, the solution works fine

**Comment by wamberg** [comment]
Hi @jezdez 

> Hey all, to move this forward it's best to just open PRs that completely pass the tests.

Please see the attached PR when you have a moment. There was an additional change for Django 3.2 compatibility other than the cache change. Tests look good on the PR :+1:

**Comment by afedosenko** [comment]
Any news?

**Comment by mpasternak** [comment]
Hi @afedosenko , I have contacted @jezdez as I need this package up and running; if @jezdez won't be able to answer, I'll create my maintained fork and keep it up as long as the original maintainers won't be available.

**Comment by mpasternak** [comment]
.... and I'm done, even if I merged pull request #118 and broke the build because of flake8, I'm unable to fix it because it says it needs reviewer. 

So either @jezdez comes back or we should move to another repo.

**Comment by hugovk** [comment]
PR #118 is merged and @mpasternak was added to the project in https://github.com/jazzband/django-dbtemplates/issues/121#issuecomment-1206571843 so we should be able to get a release out soon 🤞 

So let's close this.
